lib/api/src/externals/function.rs

```diff
@@ -87,23 +88,41 @@ impl Function {
     where
         F: Fn(&[Val]) -> Result<Vec<Val>, RuntimeError> + 'static,
     {
-        let dynamic_ctx = VMDynamicFunctionContext::from_context(VMDynamicFunctionWithoutEnv {
-            func: Box::new(func),
-            function_type: ty.clone(),
-        });
+        let dynamic_ctx: VMDynamicFunctionContext<VMDynamicFunctionWithoutEnv> =
```

Comment:
Visually the type here is redundant with the same two types written on the next line. Does rustc need this?

Reply:
This code is high risk for very bad bugs as it's doing a lot of nasty low level stuff.  I explicitly adding the type did actually fix a bug here before I believe, but I think it's best just to be extremely explicit to the compiler and the reader what the precise types of everything is here.

The bug before was that I was calling `clone` but that `clone` was actually going to `&T` and cloning the reference, not the `T` itself.

I agree that it's redundant, but I think it's probably fine to be excessively explicit when doing dangerous low-level things